core/orchestrator.go

```
@@ -315,16 +316,34 @@ func (n *LivepeerNode) transcodeSeg(config transcodeConfig, seg *stream.HLSSegme
 
 	// Prepare the result object
 	var tr TranscodeResult
+	segHashes := make([][]byte, len(tData))
+	segHashLock := &sync.Mutex{}
+
 	for i, _ := range md.Profiles {
 		if tData[i] == nil {
 			glog.Errorf("Cannot find transcoded segment for %v", seg.SeqNo)
 			continue
 		}
 		tProfileData[md.Profiles[i]] = tData[i]
 		tr.Data = append(tr.Data, tData[i])
+
+		hash := crypto.Keccak256(tData[i])
+		segHashLock.Lock()
+		segHashes[i] = hash
+		segHashLock.Unlock()
 	}
 	os.Remove(fname)
 	tr.OS = config.OS
+
+	if n == nil || n.Eth == nil {
+		return &tr
+	}
+
+	segHash := crypto.Keccak256(segHashes...)
+	tr.Sig, err = n.Eth.Sign(segHash)
```

At some point it might be nice to use the `Sign` method via the `orchestrator` struct to avoid doing the on-chain checks ourselves, but we don't have access to such a struct at this point, so this is OK.

---

Agreed that this is a bit awkward, and we could work to streamline in the future!

---

Should we propagate any signing errors? That is, `tr.Sig, tr.Err = n.Eth.Sign(segHash)`